R/occDetFunc.r

```diff
@@ -216,15 +241,46 @@ occDetFunc <- function (taxa_name, occDetdata, spp_vis, n_iterations = 5000, nyr
   # Convert the regional table to these numeric versions of site names
   if(!is.null(regional_codes)){
     regional_codes$numeric_site_name <- site_match$new_site_name[match(x = as.character(regional_codes[,1]),
-                                                             table = as.character(site_match$original_site))]
+                                                                       table = as.character(site_match$original_site))]
   }
   
-  # need to get a measure of whether the species was on that site in that year, unequivocally, in zst
-  zst <- acast(occDetdata, site ~ factor(year), value.var = 'focal', max, fill = 0) # initial values for the latent state = observed state
-  nyear <- max_year - min_year + 1
+
+  
+  # if the max_year is not null, edit the zst table to add the additional years required
+  if(!is.null(max_year)){
+    
+    # check that max_year is a numeric value
+    if(!is.numeric(max_year)) stop('max_year should be a numeric value')
+
+    # need to get a measure of whether the species was on that site in that year, unequivocally, in zst
+    zst <- acast(occDetdata, site ~ factor(year), value.var = 'focal', max, fill = 0) # initial values for the latent state = observed state
+    nyear <- max_year - min_year + 1
+    
+    # if nyear is greater than the number of years due to different specification of max_year, 
+    # add on additional columns to zst so that inital values can be create for these years.
+    
+    if(nyear > ncol(zst)){
+      # work out how many columns need to be added
+      to_add <- nyear - ncol(zst)
+      zst <- cbind(zst, matrix(0, ncol = to_add, nrow = nrow(zst)))
+      # add column names
+      colnames(zst) <- 1:nyear 
+    }
+    
+    # if a value has not been selected for max_year then continue as before
+  }else{
+    
+    # record the max year
+    max_year <- max(occDetdata$year)
+
+    # need to get a measure of whether the species was on that site in that year, unequivocally, in zst
+    zst <- acast(occDetdata, site ~ factor(year), value.var = 'focal', max, fill = 0) # initial values for the latent state = observed state
+    nyear <- max_year - min_year + 1
```

Comment:
@CharlieOuthwaite in a run where you dont send max_year I dont think min_year is defined here...

Reply:
Nope, I think I'm wrong